### ISSUE TITLE
java.lang.AssertionError: unexpected concatenated term projection

### ISSUE THREAD

**Issue by omarzd** [kframework/k-legacy#1284]
I got this on kompile:

```
[Error] Internal: Uncaught exception thrown of type AssertionError.
Please file a bug report at https://github.com/kframework/k/issues
```

Then after using the --debug option, here is the top of the stack trace:

```
java.lang.AssertionError: unexpected concatenated term projection
('delta((# <thread><props>S:Set</props>C:Bag</thread>B:Bag)(.KList)))
     at org.kframework.backend.java.kil.CellCollection$Builder.concatenate
         (CellCollection.java:353)
```

And here is the rule that has the problem:

```
    rule 
        <threads>... 
            delta(<thread> <props> S:Set </props> C:Bag </thread> B:Bag) 
            =>
            <thread> <props> S </props> C </thread>
            delta(B)
        </threads>
```

I don't understand what the problem is.

**Comment by dwightguth** [comment]
Can I see the syntax declaration for delta()?

**Comment by omarzd** [comment]
```
    syntax Bag ::= delta(Bag)   [function, latex(\delta{#1})]
    syntax Exp ::= delta(Exp)   [function, latex(\delta{#1})]
```

I want to apply it on threads and on Exps inside k cells.

**Comment by dwightguth** [comment]
Can you try making the two productions have different KLabels and let me know if that solves your issue?

**Comment by omarzd** [comment]
Just to clear the confusion for myself, this is now version 3.6. 
I did this

```
    syntax Bag ::= delta(Bag)   [function, latex(\delta{#1}), klabel(deltaOnThreads)]
    syntax Exp ::= delta(Exp)   [function, latex(\delta{#1}), klabel(deltaOnExps)]
```

but I'm still getting this:

```
java.lang.AssertionError: unexpected concatenated term projection(deltaOnThreads
((# <thread><props>S:Set</props>C:Bag</thread>B:Bag)(.KList)))
```

**Comment by dwightguth** [comment]
Okay, thanks.

@andreistefanescu how hard would it be to support functions on cell collections?

**Comment by dwightguth** [comment]
Closing as no longer relevant in new KORE. @omarz1 , this will be fixed in the upcoming K 4.0 that is planned in the next several months.
